lib/mini_magick/image.rb

```diff
@@ -532,5 +532,47 @@ def layer?
       path =~ /\[\d+\]$/
     end
 
+    ##
+    # Returns a matrix of pixels from the image. The matrix is constructed as
+    # an array (1) of arrays (2) of arrays (3) of unsigned integers:
+    #
+    # 1) one for each row of pixels
+    # 2) one for each column of pixels
+    # 3) three elements in the range 0-255, one for each of the RGB color channels
+    #
+    # @example
+    #   img = MiniMagick::Image.open 'image.jpg'
+    #   pix = image.get_pixels
+    #   pix[3][2][1] # the green channel value from the 4th-row, 3rd-column pixel
+    #
+    # A rectangular region of pixels (rather than the whole image) can be
+    # retrieved by passing the appropriate coordinates as a string argument.
+    # The following, for example, gets pixels for a 40-column by 30-row area, with
+    # the top-left corner at column 10, row 5 (see ImageMagick's documentation for
+    # more image geometry options):
+    #
+    # @example
+    #   img = MiniMagick::Image.open 'image.jpg'
+    #   pix = img.get_pixels '40x30+10+5'
+    #
+    # @return [Array] Matrix of each color of each pixel
+    def get_pixels(region=nil)
+      convert = MiniMagick::Tool::Convert.new
+      convert << path
+      convert.crop(region) if region
+      convert.depth(8)
+      convert << "RGB:-"
+      pixels = convert.call.unpack("C*")
```

Comment:
If possible, I would like that we call `String\#clear` on the RGB string output after unpacking it into pixels, so that we deallocate memory that won't use anymore.

```rb
rgb_output = convert.call
pixels = rgb_output.unpack("C*")
rgb_output.clear
```